### ISSUE TITLE
UNNEST table expression does not maintain derived column list

### ISSUE THREAD

**Issue by juriad** [jOOQ/jOOQ#13184]
### Expected behavior

The following `field` method call should find the field.
```
Table<?> unnest = DSL.unnest(DSL.field("a", int[].class));
unnest.as("ids", "id").field("id")
```

### Actual behavior

It does not find it, returns null.

Compare it with table `TEST` (as defined in the MCVE) where this works:
```
TEST.as("t", "i", "v").field("i")
```
![image](https://user-images.githubusercontent.com/886346/156389160-d2987411-6715-4fbc-a73d-ab91db3a016c.png)

A workaround is to compose the field manually:
```
DSL.field(unnest.getQualifiedName().append("id"))
```

This is how it looks like in debugger:
![image](https://user-images.githubusercontent.com/886346/156387559-43b0328e-a440-484a-89c0-ea83a5f6c530.png)

### Versions

- jOOQ: 3.16.4
- Java: 11
- Database (include vendor): Postgres
- OS: Linux
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks for your report. Yeah, it seems the default column name isn't being used. The bug is a bit bigger than just that, I think. The aliasing might not work in some dialects. Will investigate, soon.

**Comment by lukaseder** [comment]
Related issue: https://github.com/jOOQ/jOOQ/issues/13230

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.5 (https://github.com/jOOQ/jOOQ/issues/13234), and 3.15.9 (https://github.com/jOOQ/jOOQ/issues/13235).

Thanks again for your report!
